### DISCUSSION TITLE
Total count of items in the object

### DISCUSSION THREAD

**Discussion by vico-gs** [halaxa/json-machine#92] [Q&A]
As we don't use the json_decode method anymore, it's very difficult to have a precise way to know how many items we have in the object. The filesize and the strlen methods are specially for one type of elements and the content-length is very approximative isn't it ?
Is there a way to count the item we have in a jsonMachine item ? Ideally it would be a method that returns the count of the object.
If it already exists, I might have missed something !

**Comment by halaxa** [comment]
Hi, README suggests this:

> ```php
> ...
> $users = Items::fromFile('500MB-users.json');
> ```
> Count the items via `iterator_count($users)`.

Does that work for you?

**Reply by vico-gs** [reply]
I guess with the fromFile it works but with the fromIterable method it's not ....

**Reply by halaxa** [reply]
I just added a test to really make sure it works even with `fromIterable`. See fa261d25231c8bfe1ea0a29da9033f575d0860a8. Are you doing it right?

**Reply by vico-gs** [reply]
Ah I guess I know why, I'm using an httpClient from symfony

`
            $jsonChunks = $this->httpClientChunks($httpClient->stream($response));
`

`
            $data = Items::fromIterable($jsonChunks, [
                    'pointer' => '/fruits',
            ]);
`

I guess this is why the iterator_count is not working...

The error message returned is

> Cannot traverse an already closed generator

**Reply by halaxa** [reply]
The problem is actually the fact that generators, that JSON Machine uses under the hood, cannot be iterated repeatedly. `iterator_count` actually iterates the whole thing again. So if you have a code that iterates the result - count it there manually.

**Reply by vico-gs** [reply]
Very interesting, thank you very much for you're quick reply ! By the way, you're library is awesome !
